File: kashyyyk/src/test/java/org/example/product/kashyyyk/KashyyykModuleTest.java
package org.example.product.kashyyyk;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertSame;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import org.junit.jupiter.api.Test;

public class KashyyykModuleTest {

    @Test
    void testModule() throws IOException {
        KashyyykModule module = new KashyyykModule();
        assertSame(5, module.calculateSomethingImportant());

        try (InputStream hello = KashyyykModuleTest.class.getResourceAsStream("testHello.txt")) {
            String message = new BufferedReader(new InputStreamReader(hello)).readLine();
            assertEquals("TEST HELLO!", message);
        }
    }
}

File: kashyyyk/src/test/kotlin/org/example/product/kashyyyk/KashyyykModuleTest.kt
package org.example.product.kashyyyk

import java.io.BufferedReader
import java.io.InputStreamReader
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class KashyyykModuleTest {

    @Test
    fun testModule() {
        val module = KashyyykModule()
        Assertions.assertSame(5, module.calculateSomethingImportant())
        KashyyykModuleTest::class.java.getResourceAsStream("testHello.txt").use { hello ->
            val message = BufferedReader(InputStreamReader(hello)).readLine()
            Assertions.assertEquals("TEST HELLO!", message)
        }
    }
}
